README.md

```diff
@@ -218,29 +218,173 @@ Authors should respect the `prefers-reduced-motion` media query by conditionally
+Host page:
+```js
+function createWidget() {
+  let widgetContainer = document.getElementById('widgetContainer');
+  let portal = document.createElement('portal');
+  portal.src = 'widget-page.html';
+  widgetContainer.append(portal);
+  portal.addEventListener('load', () => {
+    widgetContainer.hidden = false;
+  });
+
+  portal.addEventListener('click', async (e) => {
+    e.preventDefault();
+    if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
+      await portal.animate(/* ... */).finished;
+    }
+
+    await portal.activate();
+
+    // We happen to know that the widget will adopt us, but if we needed to
+    // check, we can test for |portalHost|.
+    if (!window.portalHost)
+      return;
+
+    // The host is adopted now, so make any desired visual changes for being
+    // embedded.
+  });
+
+  portal.addEventListener(‘restore’, () => {
+    // The user pressed back, so undo any visual changes done to the host while
+    // it was adopted.
+    if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
+      await portal.animate(/* ... */).finished;
+    }
+  });
+  portal.addEventListener(‘activate’, () => {
+    // After the user returned from the widget page, they pressed forward. The
+    // widget page is implicitly activated again.
+    // Make any desired visual changes to the host page for being embedded.
+  });
+}
+```
+Widget page:
+```js
+// It may be that the user can access the widget by visiting directly. Check
+// whether we're in a portal so we can show an appropriate view.
+if (window.portalHost) {
+  // Show embedded view.
+} else {
+  // Show full view.
+}
+
+window.addEventListener('portalactivate', (e) => {
+  // Show full view.
+
+  // Augment the full view by showing the host in the background.
+  let predecessorContainer = document.getElementById('predecessorContainer');
+  let predecessor = e.adoptPredecessor();
+  // Note that the second time |portalactivate| is called (when the user presses
+  // forward), the predecessor will be the existing predecessor portal created
+  // the first time. We don't need to check for that here, since the following
+  // lines will essentially be no-ops in that case.
+  predecessorContainer.appendChild(predecessor);
+  predecessor.onactivate = () => {
+    // The user pressed back to return to the host page.
+    // Show embedded view.
+  };
+});
+```
+
+With this, the user can keep pressing back and forward which has the effect of dismissing and re-showing the widget. Suppose we now want to have the ability to dismiss the widget by tapping on the background of the widget. This can be done as follows:\
```

Comment:
```suggestion
With this, the user can keep pressing back and forward which has the effect of dismissing and re-showing the widget. Suppose we now want to have the ability to dismiss the widget by tapping on the background of the widget. This can be done as follows:

```